python.md

```diff
@@ -0,0 +1,1376 @@
+
+```python
+ds = te.DS('Requirename')
+```
+
+The `Files` method returns the file info set for the package:
+
+```python
+for f in te.Files(tag_name):
+    print(f.name)
+```
+
+This example shows how to iterate through a transaction set to get transaction
+elements.
+
+```python
+#!/usr/bin/python
+# Adds all package files on command line to a transaction
+# and prints out the transaction elements.
+# Usage:
+# python te.py rpm_file1.rpm rpm_file2.rpm ...
+
+import os
+import sys
+import rpm
+
+def read_rpm_header(ts, filename):
+    fd = os.open(filename, os.O_RDONLY)
+    try:
+        h = ts.hdrFromFdno(fd)
+    finally:
+        os.close(fd)
+    return h
+
+ts = rpm.TransactionSet()
+
+# Set to not verify DSA signatures.
+ts.setVSFlags(rpm._RPMVSF_NOSIGNATURES)
+
+for filename in sys.argv[1:]:
+    h = read_rpm_header(ts, filename)
+    print("Installing {0}".format(h[rpm.RPMTAG_NVR]))
+    ts.addInstall(h, filename, 'i')
+
+print("This will install:")
+for te in ts:
+    print("{0}-{1}-{2}".format(te.N(), te.V(), te.R()))
+
+ts.check()
+ts.order()
+
+print("This will install:")
+for te in ts:
+    print("{0}-{1}-{2}".format(te.N(), te.V(), te.R()))
+```
+
+The `te.py` script sets up a transaction and then prints out the elements, never
+completing the transaction. The purpose here is just to show what is in the
+transaction. The second set of printed output shows the results of the check and
+order methods, covered in the following section.
+
+
+## Checking and reordering the transaction elements
+
+After you have called `addInstall` or `addErase` for each of the packages you
+want to install, upgrade, or remove, you need to call two methods to verify the
+transaction set and order all the elements properly. These two methods are
+`check` and `order`.
+
+
+## Checking the Dependencies
+
+The check method checks the dependencies in a transaction set.
+
+```python
+unresolved_dependencies = ts.check()
+```
+
+It returns `None` if all dependencies are resolved, or a complex tuple for each
+unresolved dependency. In general, if the check method returns anything but
+`None`, you cannot perform the transaction.
+
+On a dependency failure, check returns a complex tuple of the dependency
+information in the following format:
+
+```
+((N,V,R), (reqN, reqV), needsFlags, suggestedPackage, sense)
+```
+
+The first element is a tuple of the name, version, and release of the package
+you are trying to install. The next tuple holds the required name and required
+version or conflicting name and version. The version will be `None` if the
+dependency is a shared library or other file.
+
+The `needsFlags` tells you about the requirement or conflict. The value is a
+bitmask that can contain the following bit settings: `rpm.RPMSENSE_EQUAL`,
+`rpm.RPMSENSE_GREATER`, and `rpm.RPMSENSE_LESS`. This tells you if the
+dependency is for a version of a package greater than 4.1, for example.
+
+The `suggestedPackage` names a package that solves the dependency. The
```

Comment:
Rpm will only suggest packages if there's a suggestion rpmdb available and enabled, which is ... never.